openfisca_core/variables.py

```diff
@@ -146,7 +147,7 @@ class Variable(object):
            Free multilines text field describing the variable context and usage.
     """
 
-    def __init__(self, baseline_variable = None):
+    def __init__(self, baseline_variable = None, enforce_no_unexpected_attr = False):
```

Comment:
> Adding this parameter allows to keep the current behavior of checking for unexpected parameters in variables created by users, but relaxes this constraints by default for easier testing.

Reply:
I would take this as a hint that the original behavior was not a good idea in the first place. Instead of adding complexity to counter complexity, consider removing the original complexity, i.e. removing the check for extra attributes. What was the value of this check in the first place?

Reply:
The value is that, as someone who models the legislation, if I mistype `defaut_value = 1` instead of `default_value = 1`, OpenFisca raises an immediate error message, rather than silently ignoring this unexpected input. 

Now, clearly having this check in the constructor was not a good idea

Reply:
Change reverted and test adapted